index.js

```diff
@@ -1,8 +1,87 @@
 const battleship = () => {
-  return 'The winner is...?'
+
+  // create objects with the given parametrs
+  let playerOne = {
+    name: "playerOne",
+    shipCount: 0,
+    gameBoard: [
+      [0, 0, 0, 0], 
+      [0, 0, 0, 0], 
+      [0, 0, 0, 0], 
+      [0, 0, 0, 0]
+    ]
+  }
+  let playerTwo = {
+    name: "playerTwo",
+    shipCount: 0,
+    gameBoard: [
+      [0, 0, 0, 0], 
+      [0, 0, 0, 0], 
+      [0, 0, 0, 0], 
+      [0, 0, 0, 0]
+    ]
+  };
+
+  // asking for the players' name
+  playerOne.name = prompt("Player 1 what is your name?");
+  playerTwo.name = prompt("Player 2 what is your name?");
+  
+  // randomly add ships to each board
+  const shipPlace = (player) => {
+    let x = Math.floor(Math.random() *4)
+      let y = Math.floor(Math.random() *4)
+      if (player.gameBoard[x][y] === 1) {
+          return
+      } else {
+        player.gameBoard[x][y]= 1;
+        player.shipCount++;
+      } 
+    }
+
+  while(playerOne.shipCount < 4) {
+    shipPlace(playerOne);
+  }
+
+  while(playerTwo.shipCount < 4) {
+    shipPlace(playerTwo);
+  }
```

Comment:
This while loops are doing the same thing. We could create a functions and pass playerOne and playerTwo as arguments (not mandatory, personal preference).

Reply:
@andresochoa91, will try it. Thank you!